### PR TITLE
Add AmpStoryCssTransformer

### PR DESCRIPTION
**Pull request by erwinmombay**
list of amp-story optimizations:
- append link[rel=stylesheet] to amp-story-1.0.css
- modify the amp-custom css to use --amp-story-${vh/vw/vmin/vmax}
- append inline <script> for the dvh polyfill
- SSR `data-story-supports-landscape`
- SSR `aspect-ratio` into style

update reoorder head transformer

### CODE REVIEW

**Comment by erwinmombay** [comment]
@sebastianbenz @westonruter would you two be able to review this. I dont have access to assign

**Comment by sebastianbenz** [comment]
I'll take a look later

File: packages/optimizer/spec/transformers/experimental/AmpStoryCssTransformer/config.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "optimizeAmpStory": true
```
Comment (erwinmombay) [inline packages/optimizer/spec/transformers/experimental/AmpStoryCssTransformer/config.json:2]: @sebastianbenz am i able to create config.json inside an individual test folder. I wanted to test one scenario for LTS

File: packages/optimizer/lib/transformers/ServerSideRendering.js [hunk 2]
```diff
@@ -115,7 +115,7 @@ class ServerSideRendering {
 
     let customStyles;
     for (let node = head.firstChild; node; node = node.nextSibling) {
-      // amp-experiment is a render delaying extension iff the tag is used in
+      // amp-experiment is a render delaying extension if the tag is used in
```
Comment (sebastianbenz) [inline packages/optimizer/lib/transformers/ServerSideRendering.js:118]: https://en.wikipedia.org/wiki/If_and_only_if

**Review by sebastianbenz — APPROVED** [review]
Added a few questions, but looks good!

One suggestion: not sure what the status if of validator support, but if support is there, it'd be good to add a validation test file here: https://github.com/ampproject/amp-toolbox/tree/main/packages/optimizer/spec/valid-amp/files

File: packages/optimizer/spec/transformers/experimental/AmpStoryCssTransformer/config.json (hunk 1 above)
Comment (sebastianbenz) [reply packages/optimizer/spec/transformers/experimental/AmpStoryCssTransformer/config.json:2]: For individual test file you can add an inline comment like here: https://github.com/ampproject/amp-toolbox/blob/main/packages/optimizer/spec/transformers/valid/RewriteAmpUrls/adds_lts/input.html#L1

File: packages/optimizer/lib/transformers/AmpStoryCssTransformer.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+  }
+}
+
+function appendAmpStoryCssLink(host, head) {
+  const ampStoryCssLink = createElement('link', {
+    'rel': 'stylesheet',
+    'amp-extension': 'amp-story',
+    'href': `${host}/v0/amp-story-1.0.css`,
+  });
+  appendChild(head, ampStoryCssLink);
```
[the remark this answers is not in the document]
Comment (erwinmombay) [reply packages/optimizer/lib/transformers/AmpStoryCssTransformer.js:144]: from my understanding, that would lock the document + css together and the stories team updates their css pretty regularly so they would prefer linking to it instead

File: packages/optimizer/lib/transformers/AmpStoryCssTransformer.js [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+const AMP_STORY_DVH_POLYFILL_CONTENT =
+  '"use strict";if(!self.CSS||!CSS.supports||!CSS.supports("height:1dvh")){function e(){document.documentElement.style.setProperty("--story-dvh",innerHeight/100+"px","important")}addEventListener("resize",e,{passive:!0}),e()})';
+
+const ASPECT_RATIO_ATTR = 'aspect-ratio';
+
+class AmpStoryCssTransformer {
+  constructor(config) {
+    this.log_ = config.log.tag('AmpStoryCssTransformer');
+
+    this.enabled_ = !!config.optimizeAmpStory;
```
[the remark this answers is not in the document]
Comment (erwinmombay) [reply packages/optimizer/lib/transformers/AmpStoryCssTransformer.js:29]: yes it should be disabled by default

File: packages/optimizer/lib/transformers/ReorderHeadTransformer.js [hunk 5]
```diff
@@ -159,6 +168,10 @@ class HeadNodes {
         this._linkStyleAmpRuntime = node;
         return;
       }
+      if (node.attribs.href.endsWith('/amp-story-1.0.css')) {
```
[the remark this answers is not in the document]
Comment (erwinmombay) [reply packages/optimizer/lib/transformers/ReorderHeadTransformer.js:171]: yes, this should be OK.

**Comment by erwinmombay** [comment]
@sebastianbenz did the PR get merged? i hadnt pushed my updates yet. will follow up on separate PR!

**Pull request merged**